xdsl/ir/core.py

```diff
@@ -756,6 +756,14 @@ def walk(self) -> Iterator[Operation]:
         for region in self.regions:
             yield from region.walk()
 
+    def walk_reverse(self) -> Iterator[Operation]:
+        """
+        Iterate all operations contained in the operation (including this one) in reverse order.
+        """
+        for region in self.regions[::-1]:
```

Comment:
Ah my bad, your lower use of `reversed` is better due to returning an iterator as opposed to a new list

Reply:
Right. I use reversed.